tests/test_executor.py

```diff
@@ -98,8 +98,8 @@ def test_missing_executable(execparams: ExecutorTestParams) -> None:
         status = job.wait(timeout=_get_timeout(execparams))
         assert status is not None
         assert status.state == JobState.FAILED
-        assert status.exit_code is not None
-        assert status.exit_code != 0
+        if status.exit_code is not None:
```

Comment:
That's a different test, this one tests what happens when you run '/bin/no_such_file_or_directory'. In that case Flux throws an exception rather than returning the exit code. It is possible (with a little more work) to fetch the exit code (probably -127) from the job, but I thought it was reasonable not to. Let me know what you think though and I can add it.

Reply:
My bad. Yeah, I think that's reasonable. Ignore my initial comment.